### PR TITLE
3.0: Fixing integration tests (scaling, slurm, fsx)

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
See commit descriptions for details.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by enrico-usai — DISMISSED** [review]

File: tests/integration-tests/configs/pcluster3.yaml [hunk 1]
```diff
@@ -6,25 +6,39 @@ test-suites:
       dimensions:
         - regions: ["af-south-1"]
           instances: {{ common.INSTANCES_DEFAULT_X86 }}
-          oss: {{ common.OSS_ONE_PER_DISTRO }}
+          oss: ["alinux2"]
```
Comment (demartinofra) [inline tests/integration-tests/configs/pcluster3.yaml:9]: why this change?

File: tests/integration-tests/configs/pcluster3.yaml [hunk 2]
```diff
@@ -6,25 +6,39 @@ test-suites:
...
           schedulers: ["slurm"]
   dns:
     test_dns.py::test_hit_no_cluster_dns_mpi:
       dimensions:
         - regions: ["eu-west-2"]
           instances: {{ common.INSTANCES_DEFAULT_X86 }}
-          oss: {{ common.OSS_COMMERCIAL_X86 }}
+          oss: ["centos7"]
```
Comment (demartinofra) [inline tests/integration-tests/configs/pcluster3.yaml:16]: why this change?

File: cli/src/pcluster/models/cluster_config.py [hunk 3]
```diff
@@ -290,6 +290,14 @@ def __init__(
         self.drive_cache_type = Resource.init_param(drive_cache_type)
         self.fsx_storage_type = Resource.init_param(fsx_storage_type)
 
+    def get_drive_cache_type(self):
+        """Drive cache type must be set for HDD (Either "NONE" or "READ"), and must not be set for SDD (None)."""
+        if self.fsx_storage_type == "HDD":
+            if self.drive_cache_type:
+                return self.drive_cache_type
+            return "NONE"
```
Comment (enrico-usai) [inline cli/src/pcluster/models/cluster_config.py:298]: This is a CFN specific logic. I mean "NONE" is a value accepted by CFN/CDK it's not a meaningful value for the model.
We should move this logic in the cluster_stack.py

Then let's be sure to have a validator that accepts a value for self.drive_cache_type only if fsx_storage_type is HDD

Comment (hanwen-pcluste) [reply cli/src/pcluster/models/cluster_config.py]: Moved. The validation was already in place

File: tests/integration-tests/configs/pcluster3.yaml (hunk 1 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/configs/pcluster3.yaml:9]: Just changed it in pcluster3.yaml. I have been trying to use pcluster3.yaml to do some quick verification. This change is to reduce the execution time of the test.

**Review by enrico-usai — APPROVED** [review]
